services/workflows-service/src/auth/auth.service.ts

```diff
@@ -11,20 +12,46 @@ export class AuthService {
     private readonly passwordService: PasswordService,
   ) {}
 
+  async validateUserById(userId: string): Promise<UserInfo | null> {
+    const user = await this.userService.getByIdUnscoped(userId, {
+      select: {
+        email: true,
+        status: true,
+        id: true,
+        firstName: true,
+        lastName: true,
+        roles: true,
+      },
+    });
+
+    if (user) {
+      return this.createUserInfo(user, user.email);
+    }
+
+    return null;
+  }
+
   async validateUser(email: string, password: string): Promise<UserInfo | null> {
     const user = await this.userService.getByEmailUnscoped(email);
 
     if (user && (await this.passwordService.compare(password, user.password))) {
-      if (user?.status !== UserStatus.Active) {
-        throw new UnauthorizedException('Unauthorized');
-      }
-
-      const { id, firstName, lastName, roles } = user;
-      const roleList = roles as string[];
-
-      return { id, email, firstName, lastName, roles: roleList };
+      return this.createUserInfo(user, email);
     }
 
     return null;
   }
+
+  private createUserInfo = (
+    user: { status: UserStatus; id: string; firstName: string; lastName: string; roles: JsonValue },
+    email: string,
+  ) => {
+    if (user?.status !== UserStatus.Active) {
```

Comment:
checking if the user active dosent seem to be the job of "createUserInfo" there is a mix up with the other function validateUserById (probably)